sklearn/impute/tests/test_impute.py

```diff
@@ -1001,6 +1002,40 @@ def test_iterative_imputer_skip_non_missing(skip_complete):
         assert_allclose(X_test_est[:, 0], [11, 7, 12], rtol=1e-4)
 
 
+@pytest.mark.parametrize(
+    "rs_imputer, rs_estimator, rs_expect",
+    [(None, None, None), (401, None, 401),
+     (np.random.RandomState(seed=402), None, "RandomState"),
+     (None, 403, 403), (404, 405, 405),
+     (np.random.RandomState(seed=406), 407, 407),
+     (None, "string", "string"), (408, "string", "string"),
+     (np.random.RandomState(seed=409), "string", "string")]
+)
+def test_iterative_imputer_set_estimator_random_state(
+    rs_imputer, rs_estimator, rs_expect
+):
+    class ZeroPredictor(BaseEstimator):
+        def __init__(self, random_state=None, rs_expect="None"):
+            self.rs_expect = rs_expect
+            self.random_state = random_state
+
+        def fit(self, X, y):
+            if self.rs_expect == "RandomState":
+                assert isinstance(self.random_state, np.random.RandomState)
```

Comment:
can't you check the identity of the RandomState?

Reply:
It's not possible unfortunately as they contain information about memory address. So something like this would return `False`:
```python
import numpy as np
np.random.RandomState(100) == np.random.RandomState(100)
>>> False
```

Reply:
We can try to check for the state explicitly:

```py
import numpy as np
from numpy.testing import assert_array_equal
rng1 = np.random.RandomState(100)
rng2 = np.random.RandomState(100)

rng1_state = rng1.get_state()
rng2_state = rng2.get_state()

assert rng1_state[0] == rng2_state[0]
assert_array_equal(rng1_state[1], rng2_state[1])
assert_array_equal(rng1_state[2:], rng2_state[2:])
```